js/PaymentRequest/index.js

```diff
@@ -258,7 +269,36 @@ export default class PaymentRequest {
         INTERNAL_SHIPPING_ADDRESS_CHANGE_EVENT,
         this._handleShippingAddressChange.bind(this)
       );
+
+      this._paymentMethodChangeSubscription = DeviceEventEmitter.addListener(
+          INTERNAL_PAYMENT_METHOD_CHANGE_EVENT,
+          this._handlePaymentMethodChange.bind(this)
+      );
+    }
+  }
+
+  _handlePaymentMethodChange(paymentMethod: PaymentMethod) {
+    this._paymentMethod = paymentMethod.paymentMethodType;
```

Comment:
If we treated `paymentMethod` as a string, we could simplify this to:
```suggestion
  _handlePaymentMethodChange(paymentMethod: string) {
    this._paymentMethod = paymentMethod;
```

Reply:
paymentMethod is handling the response body

Reply:
paymentMethod is handling the response body